Add converter debug tool (#686)
kyma-project/infrastructure-manager: hack/patch-converter-runner/README.md

# Patch Converter Runner 

## Overview

The `patch-converter-runner` program invokes converter library to generate shoot from runtime resources. The purpose of this application is to allow developers to:
- quickly reproduce problems with updating existing Gardener clusters
- debug converter without a need to run the whole KIM system

## Building the Binary

To build the `patch-converter-runner` binary navigate to the `hack` folder, and execute the following command:

```sh

>   go build -o ./bin/patch-converter-runner ./patch-converter-runner/cmd/main.go

Reviewer: This results to check for consistency of the main vendor directory of the project. 

`go: inconsistent vendoring in /Users/I523107/GoPath/src/github.com/kyma-project/infrastructure-manager:
`
 I suggest running build command inside patch-converter-runner directory: 
```suggestion
binary navigate to the `hack/patch-converter-runner` folder, and execute the following command:

  go build -o ./bin/patch-converter-runner ./cmd/main.go
```

```

## Usage

### Options

- `--runtime-path`: The path to the runtime CR file
- `--shoot-path`: The path to the existing shoot CR file
- `--kubeconfig-path`: The path to the KCP kubeconfig file
- `--outputPath`: The path to the output shoot CR file

### Example

To run the `patch-converter-runner` program, execute the following command:
```sh

> patch-converter-runner -runtime-path=<runtime CR fie path> -shoot-path=<shoot CR path> -kubeconfig-path=<KCP kubeconfig path> -outputPath=<output shoot CR path>

Reviewer: Maybe the `outputPath` is not necessary and can be just replaced with usage for the current directory

Author: I decided to leave it as it is

```

### Helper script

You can also run `./script/run.sh` script to execute the application. Before running the script you need to setup the following environment variables:
- `KCP_KUBECONFIG` - the path to the KCP kubeconfig file
- `GARDENER_KUBECONFIG` - the path to the Gardener kubeconfig file

> - `$GARDENER_NAMESPACE` - the namespace where the shoot resources are stored

Reviewer: ```suggestion
- `GARDENER_NAMESPACE` - the namespace where the shoot resources are stored, you can get it from your Gardener My Account page ex: https://dashboard.garden.live.k8s.ondemand.com/account
```

To run the script:
1. build the `patch-converter-runner` application
2. navigate to the `script` folder
3. execute the following command:
```sh

> ./script/run.sh <output folder> <runtime ID> <shoot name>

Reviewer: Remove point 2 or

```suggestion
./run.sh <output folder> <runtime ID> <shoot name>
```

Author: It should stay as it is since I modified the description

# Patch Converter Runner 

## Overview

The `patch-converter-runner` program invokes converter library to generate shoot from runtime resources. The purpose of this application is to allow developers to:

> - quickly reproduce problems with updating existing Gardener clusters

Reviewer 2: ```suggestion
- quickly reproduce problems by updating existing Gardener clusters
```

- debug converter without a need to run the whole KIM system

## Building the Binary

To build the `patch-converter-runner` binary navigate to the `hack/patch-converter-runner` folder, and execute the following command:

```sh
  go build -o ./bin/patch-converter-runner ./cmd/main.go
```

## Usage

### Options

- `--runtime-path`: The path to the runtime CR file
- `--shoot-path`: The path to the existing shoot CR file
- `--kcp-kubeconfig-path`: The path to the KCP kubeconfig file
- `--output-path`: The path to the output shoot CR file

### Example

To run the `patch-converter-runner` program, execute the following command in the `hack/patch-converter-runner` folder:
```sh
./bin/patch-converter-runner -runtime-path=<runtime CR fie path> -shoot-path=<shoot CR path> -kcp-kubeconfig-path=<KCP kubeconfig path> -output-path=<output shoot CR path>
```

To run the `patch-converter-runner` program with `go run`, execute the following command in the `hack/patch-converter-runner` folder:
```sh
go run ./cmd/main.go -runtime-path=<runtime CR fie path> -shoot-path=<shoot CR path> -kcp-kubeconfig-path=<KCP kubeconfig path> -output-path=<output shoot CR path>
```

### Helper script

> You can also run `./script/run.sh` script to execute the application. Before running the script you need to setup the following environment variables:

Reviewer 2: ```suggestion
You can also run `./script/run.sh` to execute the application. Before running the script, you need to set up the following environment variables:
```
